src/sema/functions.rs

```diff
@@ -569,24 +572,24 @@ pub fn contract_function(
                 ));
                 return None;
             }
-        } else {
-            let payable = fdecl.is_payable();
+        }
```

Comment:
Why is this moved out of the else branch of the if statement? Checking that all constructors are payable/not payable only applies if there are mulitple constructors

Reply:
Yeah I can't remember why I did that, must be a mistake. Thanks!